pkg/package-server/server/server.go

```diff
@@ -239,3 +290,24 @@ func (o *PackageServerOptions) Run(ctx context.Context) error {
 
 	return err
 }
+
+func (a *Operator) syncOLMConfig(obj interface{}) error {
+	olmConfig, ok := obj.(*operatorsv1.OLMConfig)
+	if !ok {
+		return fmt.Errorf("casting OLMConfig failed")
+	}
+	// restart the pod on change
+	if olmConfig.Spec.Features == nil || olmConfig.Spec.Features.PackageServerSyncInterval == nil {
+		if a.options.WakeupInterval != DefaultWakeupInterval {
+			log.Warnf("Change to olmConfig: '%v' != default '%v'", a.options.WakeupInterval, DefaultWakeupInterval)
+			os.Exit(0)
```

Comment:
@stevekuznetsov not sure I like how this causes a restart, but I couldn't figure out a way to reconfigure or restart the `sourceProvider`

Reply:
Yeah, thankfully this is a very commonplace way to achieve this in k8s so I think you're good